### ISSUE TITLE
Investigate issues with nested code blocks in wd.io markdown

### ISSUE THREAD

**Issue by sourishkrout** [stateful/runme#61]
Observed issues with this markdown: https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md

Please make recommendations for follow-up actions.

**Comment by adambabik** [comment]
TLDR; Categories of issues:
- Miss representation of code blocks as markdown.
- Possibly related issues with incorrect lang guesses.
- Issues with unnesting blocks, most commonly incorrect retain of white spaces prefix.

---

![Screenshot 2022-11-17 at 14 40 32](https://user-images.githubusercontent.com/277870/202461432-01046af6-f511-4ccb-a0f7-2b23cbe127fc.png)

There are a few issues here: 
- This block is reinterpreted in the extension as Markdown. I haven't found yet why because `runme json` returns `"executable": "sh"`.
- It is incorrectly rendered due to whitespaces before the closing backticks:
    ````
    ```sh
        $ git clone git@github.com:<your-username>/webdriverio.git
        ```
    ````
    This can be fixed in the parser. The opening backticks also have a prefix with spaces but it is removed.

![Screenshot 2022-11-17 at 14 51 18](https://user-images.githubusercontent.com/277870/202463838-0a1f0ae1-633f-4226-9c1f-6bcb941b9afc.png)

This is a problem with nested blocks. The top-level block, a list, was kinda split correctly but the nested blocks' (a list item) source retained prefixing white spaces. And prefixing a paragraph with at least four spaces in Markdown tells it to interpret it as code (not a fenced code block but regular code block). This is to be fixed in the parser.

![Screenshot 2022-11-17 at 14 56 15](https://user-images.githubusercontent.com/277870/202465118-b8f6900f-7c60-4617-bd73-53388b86f5d7.png)

Original source:

````
- *Dependency Checks*
  We automatically check if every sub-package has all the dependencies from its `package.json` installed.
  You can manually trigger this check by calling:
  ```sh
  $ npm run test:depcheck
  ```
````

Again, the code block is interpreted as a markdown block. The new line after `* [...]*` is actually dismissed in any Markdown renderer so it's not an error.

![Screenshot 2022-11-17 at 14 59 35](https://user-images.githubusercontent.com/277870/202466040-6f47e84b-1216-499a-b60b-46ccf45d7570.png)

Mixed results from the language detector.

**Comment by sourishkrout** [comment]
Is this done @adambabik? I suppose with the parser/renderer rewrite we'll be starting over on testing for edge cases?

**Comment by adambabik** [comment]
@sourishkrout yeah, we need to verify these edge cases visually again. After that I will rebuild acceptance tests for deserializer which are gone in my development branch as the output changed significantly.

I have a bunch of unit and integration tests and verified the most problematic files using `runme fmt`. There are minor visual differences which I started collecting in https://github.com/stateful/runme/issues/66. However, `runme fmt` is a part of equation as the complete flow works as follows: byte array --> AST --> Intermediate Representation (it uses the same functions as `runme fmt`) --> cells. We don't have enough verification of the IR --> cells tranformation yet. Problems described here are a combination of errors from the IR and IR --> cells.

**Comment by adambabik** [comment]
I am closing this issue as done. The file renders correctly using https://github.com/stateful/vscode-runme/pull/81/commits/2169911c39f1a18af0592992f9840a44415dad9d. Using `runme fmt` yields a bunch of differences but most of them is covered in https://github.com/stateful/runme/issues/66.
